benchmarking/README.md

```diff
@@ -36,3 +36,78 @@ python examples/offline_inference_tt.py --measure_perf --max_seqs_in_batch 32 --
 - `--max_seqs_in_batch` (default: `32`):
   - **Maximum batch size** for inference, determining the number of prompts processed in parallel.
 
+### Online Benchmarking
+
+#### single user
+
+```bash
+python utils/prompt_client_cli.py \
+    --num_prompts 32 \
+    --batch_size 1 \
+    --tokenizer_model meta-llama/Llama-3.1-70B-Instruct \
+    --max_prompt_length 128 \
+    --input_seq_len 128 \
+    --output_seq_len 128 \
+    --template chat_template \
+    --dataset random
+```
+
+#### using vllm/benchmarking/benchmark_serving.py
+Within the Docker container, use the benchmark_serving.patch file:
+```
+cd ~/app/src
+python run_vllm_api_server.py
+```
+This simply stops the benchmarking script from sending the `best_of` arg which is not supported and causes issues.
+
+To run the benchmarks, in another shell into the Docker container:
+```
+cd ~/vllm
+git apply ~/app/benchmarking/benchmark_serving.patch
+cd ~/app
+export PYTHONPATH=$PYTHONPATH:$PWD
+python benchmarking/vllm_online_benchmark.py
+```
+
+The output will be available for each input/output sequence length defined and time stamped.
+
+Results are also printed to stdout, for example with mock data results:
+```
+==================================================
+                    Benchmark Result                     
+==================================================
+Successful requests:                     32
+Benchmark duration (s):                  0.39
+Total input tokens:                      4096
+Total generated tokens:                  64
+Request throughput (req/s):              83.04
+Output token throughput (tok/s):         166.07
+Total Token throughput (tok/s):          10794.77
+--------------------------------------------------
+               Time to First Token                  
+--------------------------------------------------
+Mean TTFT (ms):                          358.26
+Median TTFT (ms):                        358.45
+P99 TTFT (ms):                           361.67
+--------------------------------------------------
+     Time per Output Token (excl. 1st token)       
```

Comment:
From their reporting, what is different between TPOT and ITL?

Reply:
TPOT is the average ITL over the request. Only TTFT, ITL, and latency (request latency) are raw measurements made. TPOT is calculated in https://github.com/tenstorrent/vllm/blob/dev/benchmarks/benchmark_serving.py#L311